transmart-core-db/src/main/groovy/org/transmartproject/db/tree/TreeNode.groovy

```diff
@@ -268,4 +276,26 @@ class TreeNode {
         }
     }
 
+    final String getDimension() {
+        switch (tableName) {
+            case 'concept_dimension':
+                return DimensionImpl.CONCEPT.name
+            case 'patient_dimension':
+                return DimensionImpl.PATIENT.name
+            case 'modifier_dimension':
+                def dimension = DimensionDescription.createCriteria().list {
```

Comment:
I am not really in favour of doing domain class queries at marshalling time. This will create a lot of queries and slows down marshalling unnessecarily. The lookup could also be done in the `TreeService`, enabling also caching of the dimension name for modifiers.